django_toolkit/fallbacks/circuit_breaker/rules.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+import abc
+import logging
+
+logger = logging
+
+
+class Rule(object):
+    __metaclass__ = abc.ABCMeta
+
+    def __init__(
+        self,
+        max_failures,
+        failure_cache_key,
+    ):
+        self.failure_cache_key = failure_cache_key
+        self.max_failures = max_failures
+
+    @abc.abstractmethod
+    def should_open_circuit(self, total_failures, percentage_failures=None):
+        pass
+
+    @abc.abstractmethod
+    def should_increase_failure_count(self):
+        pass
+
+    @abc.abstractmethod
+    def should_increase_request_count(self):
+        pass
+
+    def log_increase_failures(self, total_failures, percentage_failures=None):
+        logger.info(
+            'Increase failure for: {key} - '
+            'max failures {max_failures} - '
+            'total failures {total_failures}'.format(
+                key=self.failure_cache_key,
+                max_failures=self.max_failures,
+                total_failures=total_failures
+            )
+        )
+
+
+class MaxFailuresRule(Rule):
+
+    def should_open_circuit(self, total_failures, percentage_failures=None):
+        return total_failures >= self.max_failures
+
+    def should_increase_failure_count(self):
+        return True
+
+    def should_increase_request_count(self):
+        return False
+
+
+class PercentageFailuresRule(Rule):
+
+    def __init__(
+        self,
+        max_failures,
+        failure_cache_key,
+        max_accepted_failures,
```

Comment:
I don't think this parameter is necessary here, as this rule should open the circuit breaker only when the percentage is exceeded.

Reply:
Only when the percentage is exceeded and number of failures greater than the value defined in the class instance, right? Does [this rule](https://github.com/luizalabs/django-toolkit/pull/36/files#diff-cbd5e9945838021f132d5c187def585bR73) make sense to you?

Reply:
I think the `PercentageFailuresRule` should only open the circuit when the percentage is greater than the threshold.

Reply:
You're right. I confused the rules